api/topology/v1/topology_api.proto

```diff
@@ -158,3 +159,29 @@ message Edge {
   // edge.
   map<string, google.protobuf.Value> metadata = 4;
 }
+
+message TopologyObject {
+  // Id is the unique identifer of the object.
+  string id = 1;
+
+  // Pb is the clutch proto object.
+  google.protobuf.Any pb = 2;
+
+  // Metadata is set by the service which produces TopologyObjects, for example k8s would extract
+  // relevent metadata that gives the TopologyAPI the ability to query against it.
+  map<string, string> metadata = 3;
+
+  // TopologyCacheAction signifies to the Topology service what to do with an incoming TopologyObject
+  //
+  // The topology service gets TopologyObjects off of the `GetTopologyObjectChannel` which is processed
+  // and stored in the topology_cache table.
+  enum TopologyCacheAction {
+    UNSPECIFIED = 0;
+    // UPSERT creates or updates items in the topology_cache table
+    UPSERT = 1;
+    DELETE = 2;
+  }
+
+  // Action denotes what the topology service should do with this object.
+  TopologyCacheAction action = 4;
```

Comment:
does this belong on the object itself? seems like it would be in a parent message used on channel with a nested topo object.

Reply:
Yeah I see what you'r saying, that makes sense let me change it up a bit.